main.tf

```diff
@@ -26,11 +26,12 @@ module "stacks" {
   terraform_workspace       = each.value.workspace
   labels                    = each.value.labels
 
-  autodeploy        = coalesce(try(each.value.settings.spacelift.autodeploy, null), var.autodeploy)
-  branch            = coalesce(try(each.value.settings.spacelift.branch, null), var.branch)
-  repository        = coalesce(try(each.value.settings.spacelift.repository, null), var.repository)
-  terraform_version = lookup(var.terraform_version_map, try(each.value.settings.spacelift.terraform_version, ""), var.terraform_version)
-  component_root    = format("%s/%s", var.components_path, coalesce(each.value.base_component, each.value.component))
+  autodeploy           = coalesce(try(each.value.settings.spacelift.autodeploy, null), var.autodeploy)
+  branch               = coalesce(try(each.value.settings.spacelift.branch, null), var.branch)
+  repository           = coalesce(try(each.value.settings.spacelift.repository, null), var.repository)
+  terraform_version    = lookup(var.terraform_version_map, try(each.value.settings.spacelift.terraform_version, ""), var.terraform_version)
+  component_root       = format("%s/%s", var.components_path, coalesce(each.value.base_component, each.value.component))
+  enable_local_preview = try(each.value.settings.spacelift.enable_local_preview, null) != null ? each.value.settings.spacelift.enable_local_preview : var.enable_local_preview
```

Comment:
@aknysh this lookup of `settings.spacelift` won't work until we've updated the utils provider to parse that setting, correct? I don't believe it will do any harm because of the `try` though and if support is added to the stack schema it should automatically just work via this update to the module, but I'd like to get your thoughts on if there anything you see wrong with that.

Reply:
the `utils` provider returns a complete map of `settings` including everything in `settings.spacelift`, we don't need to update the provider @Gowiem

Reply:
Aha I thought it did not for some reason. That's great 👍